pkg/kubectl/cmd/autoscale.go

```diff
@@ -118,62 +110,73 @@ func RunAutoscale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []
 		return cmdutil.UsageError(cmd, fmt.Sprintf("generator %q not found.", generatorName))
 	}
 	names := generator.ParamNames()
-	params := kubectl.MakeParams(cmd, names)
-	name := info.Name
-	params["default-name"] = name
 
-	params["scaleRef-kind"] = mapping.GroupVersionKind.Kind
-	params["scaleRef-name"] = name
-	params["scaleRef-apiVersion"] = mapping.GroupVersionKind.GroupVersion().String()
+	for i := range infos {
+		info := infos[i]
+		mapping := info.ResourceMapping()
+		if err := f.CanBeAutoscaled(mapping.GroupVersionKind.GroupKind()); err != nil {
+			return err
+		}
 
-	if err = kubectl.ValidateParams(names, params); err != nil {
-		return err
-	}
-	// Check for invalid flags used against the present generator.
-	if err := kubectl.EnsureFlagsValid(cmd, generators, generatorName); err != nil {
-		return err
-	}
+		name := info.Name
```

Comment:
> A refactor appears like it would be best suited in a larger, separate PR.
> 
> Thoughts?

Worrisome, but I'm ok with a followup issue linked back here.